canister/src/types.rs

```diff
@@ -977,3 +987,67 @@ fn address_handles_script_edge_case() {
         Err(InvalidAddress)
     );
 }
+
+#[test]
+fn target_difficulty() {
+    // Example found in https://en.bitcoin.it/wiki/Difficulty#How_is_difficulty_calculated.3F_What_is_the_difference_between_bdiff_and_pdiff.3F
+    assert_eq!(
+        Block::target_difficulty(
+            Network::Mainnet,
+            bitcoin::BlockHeader::u256_from_compact_target(0x1b0404cb)
+        ),
+        16307
+    );
+
+    // Mainnet block 768362.
+    // Data pulled from https://www.blockchain.com/explorer/blocks/btc/768362
+    assert_eq!(
+        Block::target_difficulty(
+            Network::Mainnet,
+            bitcoin::BlockHeader::u256_from_compact_target(386397584)
+        ),
+        35364065900457
```

Comment:
nit: Can we use underscores in the numbers for a bit better readability?

Reply:
Done.